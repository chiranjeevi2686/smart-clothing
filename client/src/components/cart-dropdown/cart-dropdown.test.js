import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CartDropdown from './cart-dropdown.component';


const defaultStore = {
    cart: {
        cartItems: [{
            id: 19,
            imageUrl: "https://i.ibb.co/mJS6vz0/blue-jean-jacket.png",
            name: "Blue Jean Jacket",
            price: 90,
            quantity: 1
        }],
        hidden: false,
        payment: null
    }
};

const mockStore = configureStore();

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('CartDropdown component', () => {

    it('should render CartDropdown', () => {

        const { asFragment, getByTestId, getByText } = render
            (<Provider
                store={mockStore(defaultStore)}
                history={mockHistoryPush}
            >
                <CartDropdown />
            </Provider>)
        expect(asFragment()).toMatchSnapshot();
        expect(getByText('GO TO CHECKOUT')).toBeInTheDocument();
        const checkoutButton = getByTestId('checkout-button');
        fireEvent.click(checkoutButton);
    })


})
