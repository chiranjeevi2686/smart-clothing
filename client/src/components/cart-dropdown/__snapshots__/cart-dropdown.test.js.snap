// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component should render CartDropdown 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ ewDQCU"
  >
    <div
      class="sc-gsTCUz fmqGKC"
    >
      <div
        class="sc-eCssSg iRTsiw"
      >
        <img
          alt="cartImage"
          class="sc-jSgupP cAHwYm"
          src="https://i.ibb.co/mJS6vz0/blue-jean-jacket.png"
        />
        <div
          class="sc-gKsewC iLwA-Dr"
        >
          <span
            class="sc-iBPRYJ ceImlv"
          >
            Blue Jean Jacket
          </span>
          <span
            class="sc-fubCfw hjMHWY"
          >
            1 * $90
          </span>
        </div>
      </div>
    </div>
    <button
      class="sc-hKgILt grWFHd"
      data-test-id="checkout-button"
    >
      GO TO CHECKOUT
    </button>
  </div>
</DocumentFragment>
`;
